photonpump/conversations.py

```diff
@@ -498,6 +631,85 @@ def _fetch_page_message(self, from_event):
         return OutboundMessage(self.conversation_id, command, data, self.credential)
 
 
+
+class IterAllEvents(ReadAllEventsBehaviour, PageAllEventsBehaviour):
+    """Command class for iterating events from all events.
+
+    Args:
+        resolve_links (optional): True if eventstore should
+            automatically resolve Link Events, otherwise False.
+        required_master (optional): True if this command must be
+            sent direct to the master node, otherwise False.
+        correlation_id (optional): A unique identifer for this
+            command.
+
+    """
+
+    def __init__(
+        self,
+        from_event: int = 0,
+        batch_size: int = 500,
+        resolve_links: bool = True,
+        require_master: bool = False,
+        direction: StreamDirection = StreamDirection.Forward,
+        credentials=None,
+        conversation_id: UUID = None,
+        only_historic: bool = False
```

Comment:
Can I ask what "only_historic" means here?

From what I can see, you're using it to mean "stop at the end of the stream" but that's the default behaviour for `iter`. It seems like maybe you want to use a volatile subscription instead?

Could you briefly describe your use-case?

Reply:
ReadStreamEventsCompleted message contains a property `is_end_of_stream` to know that all events were read. ReadAllEventsCompleted message does not have this property so I had to compare the `commit_position` with the `next_commit_position` to know if I have reached the end of all the events

We are running some data analytics so I needed the `iter` to end at some point. If I did not compare the commit_postions then it would continue polling the EventStore for events.

I added that flag so I can control whether I wanted the `iter` to end or to poll for ever.

What do you suggest?